src/libs/metadata_view.c

```diff
@@ -791,46 +910,306 @@ void connect_key_accels(dt_lib_module_t *self)
+  g_signal_connect(G_OBJECT(mi), "activate", G_CALLBACK(_menuitem_preferences), self);
+  gtk_menu_shell_append(GTK_MENU_SHELL(menu), mi);
+}
+
+void init_presets(dt_lib_module_t *self)
+{
+}
+
+void *get_params(dt_lib_module_t *self, int *size)
+{
+  *size = 0;
+  char *params = _get_current_configuration(self);
+  if(params)
+    *size =strlen(params) + 1;
+  return params;
+}
+
+int set_params(dt_lib_module_t *self, const void *params, int size)
+{
+  if(!params) return 1;
+
+  _apply_preferences(params, self);
+  _save_preferences(self);
+  return 0;
+}
+
+void gui_init(dt_lib_module_t *self)
+{
+  /* initialize ui */
+  dt_lib_metadata_view_t *d = (dt_lib_metadata_view_t *)g_malloc0(sizeof(dt_lib_metadata_view_t));
+  self->data = (void *)d;
+
+  GtkListStore *store = gtk_list_store_new(DT_METADATA_NUM_COLS,
+                                           G_TYPE_INT,      // index
+                                           G_TYPE_STRING,  // name
+                                           G_TYPE_STRING,  // displayed name
+                                           G_TYPE_STRING,  // tooltip
+                                           G_TYPE_STRING,  // value
+                                           G_TYPE_BOOLEAN, // visibility
+                                           G_TYPE_INT      // order
+                                           );
+  d->model = GTK_TREE_MODEL(store);
+
+  _lib_metadata_view_init_labels(d->model);
+
+  GtkTreeModel *sort_model = gtk_tree_model_sort_new_with_model(d->model);
+  d->sort_model = sort_model;
+
+  gtk_tree_sortable_set_sort_column_id(GTK_TREE_SORTABLE(sort_model),
+                                      DT_METADATA_COL_ORDER,
+                                      GTK_SORT_ASCENDING);
+  GtkTreeModel *filter_model = gtk_tree_model_filter_new(sort_model, NULL);
+  d->filter_model = filter_model;
+  gtk_tree_model_filter_set_visible_column(GTK_TREE_MODEL_FILTER(filter_model),
+                                           DT_METADATA_COL_VISIBLE);
+
+  GtkWidget *view = gtk_tree_view_new_with_model(GTK_TREE_MODEL(filter_model));
+  d->view = view;
+  gtk_tree_view_set_headers_visible(GTK_TREE_VIEW(view), FALSE);
+  g_object_unref(filter_model);
+
+  // metadata column
+  GtkCellRenderer *renderer = gtk_cell_renderer_text_new();
+  g_object_set(renderer, "xalign", 0.0,
+  // FIXME                         "ellipsize", PANGO_ELLIPSIZE_END,
+                         NULL);
+  // FIXME
+  //  gtk_widget_set_halign(label, GTK_ALIGN_FILL); // make it span the whole available width
+  GtkTreeViewColumn *column = gtk_tree_view_column_new_with_attributes(_("name"), renderer,
+                                        "text", DT_METADATA_COL_NAME_L, NULL);
+  gtk_tree_view_append_column(GTK_TREE_VIEW(view), column);
+
+  // value column
+  renderer = gtk_cell_renderer_text_new();
+  g_object_set(renderer, "xalign", 0.0,
+                         "ellipsize", PANGO_ELLIPSIZE_MIDDLE,
+                         NULL);
```

Comment:
This used to have
```
k == md_exif_model || k == md_exif_lens || k == md_exif_maker	
? PANGO_ELLIPSIZE_END 
: PANGO_ELLIPSIZE_MIDDLE
```
as I think for those fields middle ellipsizing doesn't work very well, but I wouldn't immediately know how to apply row specific formatting in a tree view.

Reply:
Good reminder. Done.